src/main/scala/intellij/haskell/action/DownloadCabalConfigFile.scala

```diff
@@ -0,0 +1,17 @@
+package intellij.haskell.action
+
+import com.intellij.openapi.actionSystem.{AnAction, AnActionEvent}
+import intellij.haskell.external.component.HaskellComponentsManager.refreshGlobalProjectInfo
+import intellij.haskell.external.component.PathComponent
+
```

Comment:
I understand you. Problem is that user does not see relation between auto completion of package names and cabal.config file.

I think the most general solution is to download cabal.config file when creating project (just as library sources) and extend the action which restart the repls with invalidating/refreshing everything. So when user has problems because some plugin dependency is missing or for example he changes package dependencies in Cabal file, changes Stackage LTS version or changes language extensions; he can just use one action.